sklearn/linear_model/_cd_fast.pyx

```diff
@@ -313,53 +320,50 @@ def sparse_enet_coordinate_descent(
     # that every calculation results as if we had rescaled y and X (and therefore also
     # X_mean) by sqrt(sample_weight) without actually calculating the square root.
     # We work with:
-    #     yw = sample_weight
+    #     yw = sample_weight * y
     #     R = sample_weight * residual
     #     norm_cols_X = np.sum(sample_weight * (X - X_mean)**2, axis=0)
 
+    if floating is float:
+        dtype = np.float32
+    else:
+        dtype = np.float64
+
     # get the data information into easy vars
     cdef unsigned int n_samples = y.shape[0]
     cdef unsigned int n_features = w.shape[0]
 
     # compute norms of the columns of X
-    cdef unsigned int ii
-    cdef floating[:] norm_cols_X
-
-    cdef unsigned int startptr = X_indptr[0]
-    cdef unsigned int endptr
+    cdef floating[:] norm_cols_X = np.zeros(n_features, dtype=dtype)
 
     # initial value of the residuals
     # R = y - Zw, weighted version R = sample_weight * (y - Zw)
-    cdef floating[::1] R
-    cdef floating[::1] XtA
+    cdef floating[::1] R = np.empty(n_samples, dtype=dtype)
```

Comment:
Do we need to initialize here considering we are assigning a copy of either y or yw below?

Reply:
Yes, it's C logic not Python logic. The BLAS copy function requires pointers to already allocated memory.

Reply:
```suggestion
    cdef floating[::1] R
```